### PR TITLE
Added simple unit tests for calculator

### PR DESCRIPTION
**Pull request by fromL**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -8,20 +8,50 @@
...
+        <!-- https://mvnrepository.com/artifact/joda-time/joda-time -->
+        <dependency>
+            <groupId>joda-time</groupId>
+            <artifactId>joda-time</artifactId>
+            <version>${joda-time.version}</version>
+        </dependency>
     </dependencies>
-
-
 </project>
```
Comment (OlehBEleks) [inline pom.xml]: Please add newline

File: pom.xml [hunk 2]
```diff
@@ -8,20 +8,50 @@
     <artifactId>automation-test-project</artifactId>
     <version>1.0-SNAPSHOT</version>
 
+    <properties>
+        <selenium-java.version>3.141.59</selenium-java.version>
+        <testng.version>7.3.0</testng.version>
+        <joda-time.version>2.10.6</joda-time.version>
+        <maven-surefire-plugin.version>3.0.0-M5</maven-surefire-plugin.version>
+        <suitePath>src/test/resources</suitePath>
+    </properties>
```
Comment (OlehBEleks) [inline pom.xml]: Nice job, very handy to store versions

File: src/test/resources/testng.xml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+<suite name="Test Suite 1" verbose="1">
+    <parameter name="a" value="5"/>
+    <parameter name="b" value="15"/>
+    <test name="All Tests">
+        <classes>
+            <class name = "TestCalculator"/>
+        </classes>
+    </test>
+</suite>
```
Comment (OlehBEleks) [inline src/test/resources/testng.xml]: Please add newline

File: src/test/java/TestCalculator.java [hunk 4]
```diff
@@ -0,0 +1,137 @@
+import org.junit.After;
+import org.testng.Assert;
+import org.testng.annotations.*;
+import org.testng.asserts.SoftAssert;
+
+public class TestCalculator  { // Do not extend it from parent Calculator class
```
Comment (OlehBEleks) [inline src/test/java/TestCalculator.java]: You can extend it from e.g. BaseTest that can hold some before/after methods for all tests

File: src/test/java/TestCalculator.java [hunk 5]
```diff
@@ -0,0 +1,137 @@
+import org.junit.After;
```
Comment (OlehBEleks) [inline src/test/java/TestCalculator.java]: Please remove import of junit

File: src/main/java/Calculator.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
+public class Calculator {
```
Comment (OlehBEleks) [inline src/main/java/Calculator.java]: Please add named package.

File: src/test/java/TestCalculator.java (hunk 5 above)
Comment (OlehBEleks) [inline src/test/java/TestCalculator.java]: Please add named package.

File: src/test/java/TestCalculator.java [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+    @AfterMethod
+    public void tearDown(){
+        System.out.println("This is method executed after running Method");
+    }
+
+    @Test (groups = "passedgroup", priority = 1) //Test with parameters from testng.xml
+    @Parameters ({"a", "b"})
+    public void testAdd(int x, int y){
+        int expectedResult = 20;
+        Assert.assertEquals(Calculator.add(x, y), expectedResult);
```
Comment (OlehBEleks) [inline src/test/java/TestCalculator.java]: It is recommended to add assertion messages to be more clear about the results

File: src/test/java/TestCalculator.java [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+
+    @Test (groups = "passedgroup", priority = 3, expectedExceptions = IllegalArgumentException.class)
+    public void testDivideByZero(){
+        calc.divide(calc.getA(), 0);
+    }
+
+    @Test (priority = 7)
+    public void testHardAsserts(){
+        Assert.assertEquals(2 + 2, 4);
+        System.out.println("2 + 2 = 4");
```
Comment (OlehBEleks) [inline src/test/java/TestCalculator.java]: As an option, you can add this text as assertion message so you can view only failed messages

File: src/test/java/TestCalculator.java [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+    }
+
+    @AfterMethod
+    public void tearDown(){
+        System.out.println("This is method executed after running Method");
+    }
+
+    @Test (groups = "passedgroup", priority = 1) //Test with parameters from testng.xml
+    @Parameters ({"a", "b"})
+    public void testAdd(int x, int y){
```
Comment (OlehBEleks) [inline src/test/java/TestCalculator.java]: For parameters from suite, it is recommended to add default value with @Optional
In this case you will be able to run test straight from the test file

File: src/test/resources/testng2.xml [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+        <groups>
+            <run>
+                <include name = "failedgroup"/>
+            </run>
+        </groups>
+        <classes>
+            <class name = "TestCalculator"/>
+        </classes>
+    </test>
+</suite>
```
Comment (OlehBEleks) [inline src/test/resources/testng2.xml]: Please add newline

**Review by OlehBEleks — COMMENTED** [review]
Good job 👍 
Feel free to merge after fixing minor comments

**Comment by fromL** [comment]
@OlehBEleks, thank you very much for review, I made corresponding changes and merged them into master.

**Pull request merged**